src/types/simulation.rs

```diff
@@ -84,11 +80,8 @@ impl<M: Middleware> From<SimulateValidationError<M>> for SimulationError {
             SimulateValidationError::CallStackValidation { message } => {
                 SimulationError::owned(OPCODE_VALIDATION_ERROR_CODE, message, None::<bool>)
             }
-            SimulateValidationError::Middleware(_) => {
-                SimulationError::from(ErrorCode::InternalError)
-            }
             SimulateValidationError::UnknownError { error } => {
-                SimulationError::owned(SIMULATE_VALIDATION_ERROR_CODE, error, None::<bool>)
+                SimulationError::owned(ErrorCode::InternalError.code(), error, None::<bool>)
```

Comment:
@Vid201 I am curious why unknow error belongs to `SIMULATE_VALIDATION_ERROR_CODE`.

Reply:
I changed that into internal error. But like your ideas on it.